mongodbatlas/resource_mongodbatlas_project.go

```diff
@@ -292,14 +295,62 @@ func resourceMongoDBAtlasProjectDelete(ctx context.Context, d *schema.ResourceDa
 	conn := meta.(*MongoDBClient).Atlas
 	projectID := d.Id()
 
-	_, err := conn.Projects.Delete(ctx, projectID)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"DELETING", "PENDING", "REPEATING"},
+		Target:     []string{"IDLE"},
+		Refresh:    resourceProjectClustersDeleteRefreshFunc(ctx, projectID, conn),
+		Timeout:    3 * time.Hour,
+		MinTimeout: 30 * time.Second,
+		Delay:      0,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
+
+	if err == nil {
+		_, err = conn.Projects.Delete(ctx, projectID)
+	}
+
 	if err != nil {
 		return diag.Errorf(errorProjectDelete, projectID, err)
 	}
 
 	return nil
 }
 
+func resourceProjectClustersDeleteRefreshFunc(ctx context.Context, projectID string, client *matlas.Client) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		c, resp, err := client.AdvancedClusters.List(ctx, projectID, nil)
+
+		if err != nil && strings.Contains(err.Error(), "reset by peer") {
+			return nil, "REPEATING", nil
```

Comment:
we shouldn’t use `strings.Contains` for error handling.  We should use `errors.As(err, target)`

example: 
```
	var target *matlas.ErrorResponse
	if errors.As(err, &target) && target.ErrorCode != "USER_UNAUTHORIZED" { ...
```

Reply:
I've re-worked how the error handling is done a bit, could you take a look at the latest and let me know what you think?